### DISCUSSION TITLE
How to extract images by their positions in the same page?

### DISCUSSION THREAD

**Discussion by louis-li** [pymupdf/PyMuPDF#1100] [Looking for help]
I'm extracting images from a scaned PDF. There are sometime more than 1 image on the page and I need to extract them by it's position on the page. For example, a page contains 2 images, I need to a sure way to extract the first image (on the top) first. Is there a way to achieve it? 

I'm using doc.getPageImageList(pno) and I've tried sort by xref or natural order, neither could guanrant the order. I'm wondering if I have missed anything?

**Comment by JorjMcKie** [comment]
Images on scanned PDF pages cover the complete page - obviously. Depending on the scanner hardware, the page appearance is sometimes made up by 2 image objects put upon each other, with transparency, etc.
So chances are in those case you have 2 items in `doc.get_page_images()` with identical bboxes / rectangles.
In those cases, to obtain the original image sequence, use `page.get_image_info(xrefs=True)`. This method returns a list of complete image metadata (Python dict), including object xrefs if possible. The second dictionary in that list represents the image "on top".
Image binaries are not extracted here. You can extract those separately by using the provided xref.
